plugins/modules/sda_fabric_devices_workflow_manager.py

```diff
@@ -2599,27 +2604,55 @@ def get_device_params(self, fabric_id, network_id, device_details, config_index)
                     )
                     self.status = "failed"
                     return self.check_return_status()
+            else:
+                if have_layer3_settings and (str(local_autonomous_system_number) != str(have_layer3_settings.get("localAutonomousSystemNumber"))):
+                    self.msg = (
+                        "The parameter 'local_autonomous_system_number' under 'layer3_settings' should not be "
+                        "updated for the device with IP '{ip}'.".format(ip=device_ip)
+                    )
+                    self.status = "failed"
+                    return self.check_return_status()
 
             self.validate_local_autonomous_system_number(local_autonomous_system_number, device_ip)
             self.log(
                 "Successfully validated 'local_autonomous_system_number': {asn_number}"
                 .format(asn_number=local_autonomous_system_number), "DEBUG"
             )
-            is_default_exit = layer3_settings.get("layer3_settings")
-            if not is_default_exit:
+            is_default_exit = layer3_settings.get("is_default_exit")
+            if is_default_exit is None:
                 if have_layer3_settings:
                     have_is_default_exit = have_layer3_settings.get("isDefaultExit")
                     is_default_exit = have_is_default_exit
```

Comment:
I don't see have_is_default_exit is used anywhere else..

    if have_layer3_settings:
        is_default_exit = have_layer3_settings.get("isDefaultExit", True)
    else:
        is_default_exit = True